### ISSUE TITLE
Add feature to show npm errors when publishing to npm via GitHub Actions

### ISSUE THREAD

**Issue by yashsharma999** [changesets/action#280]
Hi everyone,

I've been using Changeset to automate my releases and publishing to npm via GitHub Actions. However, I recently ran into an issue where I couldn't publish my package to npm using Changeset. Instead of getting the error message I was expecting, which I got when I tried to publish via the terminal (npm publish), I only received the following message: "**no unpublished projects to publish**".

After some investigation, I found out that the problem was with my package.json file. Specifically, I had forgotten to set **"private": false** in the file, which was preventing Changeset from publishing the package.

While I eventually figured out the problem, I think it would be helpful if it could display the actual npm error message when publishing fails, instead of the generic "no unpublished projects to publish" message. This would save time and frustration for users like me who are trying to debug issues with publishing.

Could you consider adding this feature ?

Thanks for your consideration!

**Comment by bluwy** [comment]
I don't think we necessarily want to error because some projects may have actual private packages that they do not want to publish. However, we could definitely improve the message hinting about the private field. I'd suggested opening a feature request in https://github.com/changesets/changesets if you'd like to see this as it's [handled there](https://github.com/changesets/changesets/blob/372523f4c2ee4ffeb8330d444d47ffb6d0af5126/packages/cli/src/commands/publish/index.ts#L78-L83).
